services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -49,31 +56,57 @@
   private final ScheduledExecutorService _executor;
   private final int _produceDelayMs;
   private final boolean _sync;
-  private final Map<Integer, AtomicLong> _nextIndexPerPartition;
-  private final int _partitionNum;
+  /** This can be updated while running when new partitions are added to the monitored topic. */
+  private final ConcurrentMap<Integer, AtomicLong> _nextIndexPerPartition;
+  /** This is the last thing that should be updated after adding new partitions. */
+  private final AtomicInteger _partitionNum;
   private final int _recordSize;
   private final String _topic;
   private final String _producerId;
   private final AtomicBoolean _running;
+  private final String _zkConnect;
```

Comment:
Why do we need _zkConnect and _brokerList as private variable here? Is this needed later for rebalance related code?